File: src/Web/WebApi/src/service/CategoryService.js
export default class CategoryService {
    constructor(request) {
        this.request = request;
    }

    async getList(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/category/list',
            params: filter
        });
        return response?.data ?? response;
    }

    async create(payload) {
        await this.request({
            method: 'post',
            url: '/category/create',
            data: payload
        });
    }

    async update(payload) {
        await this.request({
            method: 'post',
            url: '/category/update',
            data: payload
        });
    }

    async delete(categoryId) {
        return this.request({ method: 'delete', url: `/category/delete/${categoryId}` });
    }

    async getLookup(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/category/lookup',
            params: filter
        });
        return response?.data ?? response;
    }
}

File: src/Mobile/lib/services/category_service.dart
import 'package:dio/dio.dart';

class CategoryService {
  CategoryService(this._dio);

  final Dio _dio;

  Future<dynamic> getList([Map<String, dynamic>? filter]) async {
    final r = await _dio.get<dynamic>('/category/list', queryParameters: filter);
    return r.data;
  }

  Future<void> create(Map<String, dynamic> payload) async {
    await _dio.post<void>('/category/create', data: payload);
  }

  Future<void> update(Map<String, dynamic> payload) async {
    await _dio.post<void>('/category/update', data: payload);
  }

  Future<void> delete(int categoryId) async {
    await _dio.delete<void>('/category/delete/$categoryId');
  }

  Future<dynamic> getLookup([Map<String, dynamic>? filter]) async {
    final r = await _dio.get<dynamic>('/category/lookup', queryParameters: filter);
    return r.data;
  }
}
